src/parser/utils.ts

```diff
@@ -60,3 +61,19 @@ export function isDate(d: unknown): d is Date {
 export function isObjectLike(candidate: unknown): candidate is Record<string, unknown> {
   return typeof candidate === 'object' && candidate !== null;
 }
+
+export function deprecate<T extends Function>(fn: T, message: string): T {
+  if (typeof window === 'undefined' || typeof self === 'undefined') {
+    // eslint-disable-next-line @typescript-eslint/no-var-requires
+    return require('util').deprecate(fn, message);
```

Comment:
Is there still any benefit to using the Node built-in at this point? I like this runtime agnostic one

Reply:
If we're in nodejs this will handle the `--throw-deprecation`, `--no-deprecation`, `--trace-deprecation` flags.